src/app/ui/user-profile/user-profile.component.ts

```diff
@@ -79,6 +86,18 @@ export class UserProfileComponent implements OnInit {
     });
   }
 
+  private getOwnExp(uid: string) {
+    this.playerPointsService.getTotalExp(uid).subscribe(exp => {
+      this.ownExp = exp;
+    });
+  }
+
+  private getPlayerExp(uid: string) {
```

Comment:
this and `getOwnExp()` can be merged into one. please just return the observable and don't subscribe and set it to a property.